// @aws-sdk/client-ecr/dist-es/commands/GetRegistryScanningConfigurationCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetRegistryScanningConfigurationRequestFilterSensitiveLog, GetRegistryScanningConfigurationResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetRegistryScanningConfigurationCommand, serializeAws_json1_1GetRegistryScanningConfigurationCommand, } from "../protocols/Aws_json1_1";
export class GetRegistryScanningConfigurationCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, GetRegistryScanningConfigurationCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ECRClient";
        const commandName = "GetRegistryScanningConfigurationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: GetRegistryScanningConfigurationRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetRegistryScanningConfigurationResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1GetRegistryScanningConfigurationCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1GetRegistryScanningConfigurationCommand(output, context);
    }
}
